### PR TITLE
Fix #8: file_eq for ndiff

### LINKED ISSUE #8: Use file_eq for ndiff comparisons
Change ndiff comparison to use file_eq once pykern file_eq can do ndiff

### PR DESCRIPTION
**Pull request by gurhar1133**
now using file_eq to ndiff test

### CODE REVIEW

File: tests/lct_test.py [hunk 1]
```diff
@@ -67,11 +67,11 @@ def test_lct_abscissae():
             ]
         )
     )
-    _ndiff_files(
-        data_dir.join("lct_abscissae_outputs.txt"),
-        pkio.write_text(work_dir.join("lct_abscissae_outputs_actual.txt"), a),
-        work_dir.join("ndiff.out"),
-        data_dir,
+    pkunit.file_eq(
```
Comment (robnagler) [inline tests/lct_test.py:70]: Can't this use "actual"? The actual_path should be created in file_eq.

File: tests/lct_test.py [hunk 2]
```diff
@@ -95,23 +93,19 @@ def test_chirp_multiply():
     dus, all_fvals, all_uvals = _vals()
     sigs = list(zip(dus, all_fvals))
     r = _cast_from_complex_signal(sigs, lct.chirp_multiply, _Q_CM)
-    _ndiff_files(
-        data_dir.join("cm_signals.txt"),
-        pkio.write_text(work_dir.join("cm_signals_actual.txt"), r),
-        work_dir.join("ndiff.out"),
-        data_dir,
+    pkunit.file_eq(
+        expect_path=data_dir.join("cm_signals.ndiff"),
```
Comment (robnagler) [inline tests/lct_test.py:97]: I do wonder if this couldn't be done with case_dirs. There seems to be a lot of copy and paste here.

**Review by robnagler — CHANGES_REQUESTED** [review]

**Comment by gurhar1133** [comment]
@robnagler made changes lmk your thoughts

File: tests/lct_test.py [hunk 3]
```diff
@@ -20,12 +20,94 @@
 ]
 
 
-def test_lct_signal():
+class _MultiCases:
```
Comment (robnagler) [inline tests/lct_test.py:23]: Lambdas are harder to read. It's nice to name the difference cases as functions. I did a few things including sorting the methods. This binds the name to the case to the file. I don't think the directories needs to be named "case_x". Rather, just "x.in" and "x.out"

**Review by robnagler — CHANGES_REQUESTED** [review]
Could move all of the code into the class. Not sure what test_apply_2d_sep does.

File: tests/lct_test.py [hunk 4]
```diff
@@ -20,12 +20,97 @@
...
+    def __init__(self):
+        self._values = self._vals()
+        self._signal = self._sigs()
+        for d in pkunit.case_dirs():
+            f = getattr(self, f"_{d.basename}")
```
Comment (robnagler) [inline tests/lct_test.py:28]: Dispatch methods should always have `_case` in front of them as I had written.

**Review by robnagler — CHANGES_REQUESTED** [review]

File: tests/lct_test.py (hunk 4 above)
Comment (gurhar1133) [reply tests/lct_test.py:28]: my bad

File: tests/lct_test.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-u"""Tests for linear canonical transform functions
+"""Tests for linear canonical transform functions
```
Comment (robnagler) [inline tests/lct_test.py:2]: Use our normal header here.

File: tests/lct_test.py [hunk 6]
```diff
@@ -20,12 +20,97 @@
...
 
 
-def test_lct_signal():
+class _MultiCases:
+    def __init__(self):
+        self._values = self._vals()
+        self._signal = self._sigs()
+        for d in pkunit.case_dirs():
+            f = getattr(self, f"_case_{d.basename}")
+            pkio.write_text(
```
Comment (robnagler) [inline tests/lct_test.py:29]: Could be on one line.

File: tests/lct_test.py [hunk 7]
```diff
@@ -20,12 +20,97 @@
 ]
 
 
-def test_lct_signal():
+class _MultiCases:
+    def __init__(self):
+        self._values = self._vals()
```
Comment (robnagler) [inline tests/lct_test.py:25]: Does these need to be cached? If so, then _vals and _sigs nested functions to `__init__` and assign to self._sigs, _vals.

File: tests/lct_test.py [hunk 8]
```diff
@@ -20,12 +20,97 @@
...
+    def _vals(self):
+        dus = []
+        all_fvals = []
+        all_uvals = []
+        for i, inputs in enumerate(((3.0, 69), (5.0, 50), (8.0, 100), (8.0, 100))):
+            du, uvals, fvals = d_f_u_vals(*inputs, i)
+            dus.append(du)
+            all_fvals.append([f for f in fvals])
+            all_uvals.append([u for u in uvals])
+        return dus, all_fvals, all_uvals
```
Comment (robnagler) [inline tests/lct_test.py:71]: PKDict would be clearer.

File: tests/lct_test.py [hunk 9]
```diff
@@ -174,34 +150,6 @@ def _case(case_number, data_dir):
...
-        ],
-        output=str(diff_file),
-    )
-
-    d = pkio.read_text(diff_file)
-    if re.search("diffs have been detected", d):
-        raise AssertionError(f"{d}")
-
-
 def _cast_complex_for_write(number):
```
Comment (robnagler) [inline tests/lct_test.py:153]: The word `cast` is loaded with meaning that isn't quite what's going on here. Indeed, I think what you would want for "write" is to convert it to a string so _complex_as_str would be a better name, I think, and to change what it produces.

**Review by robnagler — CHANGES_REQUESTED** [review]

**Review by robnagler — APPROVED** [review]

File: tests/lct_test.py (hunk 5 above)
Comment (robnagler) [reply tests/lct_test.py]: This wasn't done.

Comment (gurhar1133) [reply tests/lct_test.py]: like with 
```python
u""" 
...
:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
:license: http://www.apache.org/licenses/LICENSE-2.0.html
...
"""
```
?

Comment (robnagler) [reply tests/lct_test.py]: Yes but don't need `u"""`  just `"""`. Describe what the file does on the first line.

**Comment by gurhar1133** [comment]
@robnagler good to merge now?

**Review by robnagler — APPROVED** [review]

**Pull request merged**